pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -248,6 +255,37 @@ func (p *tinkerbellProvider) DeleteResources(ctx context.Context, clusterSpec *c
 	return p.providerKubectlClient.DeleteEksaDatacenterConfig(ctx, eksaTinkerbellMachineResourceType, p.datacenterConfig.Name, clusterSpec.ManagementCluster.KubeconfigFile, p.datacenterConfig.Namespace)
 }
 
+func (p *tinkerbellProvider) PostClusterDeleteValidate(ctx context.Context, managementCluster *types.Cluster) error {
+	// We want to validate cluster nodes are powered off.
+	// We wait on BMC status.powerState to check for power off.
+	bmcRefs := make([]string, 0, len(p.hardwares))
+	for _, hw := range p.hardwares {
+		bmcRefs = append(bmcRefs, hw.Spec.BmcRef)
+	}
+
+	// TODO (pokearu): The retry logic can be substituted by changing GetBmcsPowerState to use kubectl wait --for
+	// In the current version of kubectl in EKSA --for does not support jsonpath.
+	err := retrier.Retry(10, 10*time.Second, func() error {
+		powerStates, err := p.providerKubectlClient.GetBmcsPowerState(ctx, bmcRefs, managementCluster.KubeconfigFile, constants.EksaSystemNamespace)
+		if err != nil {
+			return err
+		}
+
+		for _, state := range powerStates {
+			if !strings.Contains(state, bmcStatePowerActionHardoff) {
+				return fmt.Errorf("bmc current power state: %s expected power state: %s", state, bmcStatePowerActionHardoff)
```

Comment:
The context separation here (addition of `:`) is misleading, I think you want it to read as a single sentence. To do that, use `;`.

```suggestion
				return fmt.Errorf("bmc current power state %s; expected power state %s", state, bmcStatePowerActionHardoff)
```